lib/Models/BoxDrawing.ts

```diff
@@ -776,6 +790,38 @@ export default class BoxDrawing {
     const scratchMoveStep = new Cartesian3();
     const scratchSurfacePoint = new Cartesian3();
     const scratchSurfacePoint2d = new Cartesian2();
+    const scratchNextStep = new Cartesian3();
+    const scratchNextCartographic = new Cartographic();
+
+    const keepHeightSteady = (
+      currentPosition: Cartesian3,
+      moveStep: Cartesian3
```

Comment:
1. Find the next cartesian position (x,y,z wrt center of earth) from the `moveStep`
2. Compute its cartographic (lat, lon, height) position. 
3. Then set the cartographic height of the next position to the previous cartographic height and convert back to cartesian.